src/modules/request-recorder.js

```diff
@@ -126,8 +143,15 @@ module.exports = (opts) => {
           }) : recordings, null, 2)
       }, resolve));
     },
-    release: () => {
+    release: async () => {
       assert(nockDone !== null);
+      for (let idx = 0; idx < expectedCassette.length; idx += 1) {
+        if (expectedCassette[idx] instanceof Promise) {
+          // eslint-disable-next-line no-await-in-loop
```

Comment:
Can we do this as a map and then do a reverse + forEach? Might be cleaner and concurrent. If we can assume functions never return other functions... Otherwise we have to wrap with while...

Reply:
I like this much better... speed isn't really an issue here and requests should be made in sequence anyways (whenever that use case comes up)